.github/workflows/main.yml

```diff
@@ -93,95 +89,98 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
-          name: wheels
+          path: dist
+
+  sdist:
+    needs: [check]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+          architecture: x64
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install build
+      - name: Build sdist
+        run: |
+          # PEP 517 package builder from pypa
+          python -m build --sdist
+      - uses: actions/upload-artifact@v2
+        with:
           path: dist
 
   publish_on_pypi:
     name: Publish
-    needs: [build_for_linux, build_for_macos, build_for_windows]
+    needs: [linux, windows, macos, sdist]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/download-artifact@v2
         with:
-          name: wheels
-      - uses: actions/setup-python@v2
-        with:
-          python-version: 3.9
+          name: artifact
+          path: dist
       - name: Publish on TestPyPI
-        if: github.ref == 'refs/head/main'
-        env:
-          TWINE_USERNAME: LSchueler
-          TWINE_PASSWORD: ${{ secrets.TEST_PYPI_PASSWORD }}
-        run: |
-          pip install --upgrade twine
-          twine upload --repository testpypi --skip-existing *
+        if: github.ref == 'refs/heads/main'
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.test_pypi_password }}
+          repository_url: https://test.pypi.org/legacy/
+          skip_existing: true
       - name: Publish on PyPI
         if: startsWith(github.ref, 'refs/tags')
-        env:
-          TWINE_USERNAME: LSchueler
-          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-        run: |
-          pip install --upgrade twine
-          twine upload --skip-existing *
-
-  #publish_on_pypi:
-    #name: Publish
-    ##needs: [build_for_linux, build_for_macos, build_for_windows]
-    #needs: build_for_macos
-    #runs-on: ubuntu-latest
-    #steps:
-      #- uses: actions/download-artifact@v2
-        #with:
-          #name: wheels
-      #- name: Publish on TestPyPI
-        ##if: github.ref == 'refs/head/main'
-        #uses: pypa/gh-action-pypi-publish@release/v1
-        #with:
-          #user: LSchueler
-          #password: ${{ secrets.test_pypi_password }}
-          #repository_url: https://test.pypi.org/simple/
-          #skip_existing: true
-      #- name: Publish on PyPI
-        #if: startsWith(github.ref, 'refs/tags')
-        #uses: pypa/gh-action-pypi-publish@release/v1
-        #with:
-          #user: LSchueler
-          #password: ${{ secrets.pypi_password }}
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.test_pypi_password }}
 
   #This will probably never be needed, but who knows...
```

Comment:
Somewhat unrelated, but I think the clutter should go as this does not really provide a viable Rust API and this can be resurrected from version control in any case.

Reply:
removed.